dags/exchange_rate.py

```diff
@@ -0,0 +1,322 @@
+    return dagster.MaterializeResult(
+        metadata={
+            "date": dagster.MetadataValue.text(date_str),
+            "base_currency": dagster.MetadataValue.text("USD"),  # Always USD as per table design
+            "currencies_count": dagster.MetadataValue.int(currencies_count),
+            "min_rate": dagster.MetadataValue.float(min_rate),
+            "max_rate": dagster.MetadataValue.float(max_rate),
+            "avg_rate": dagster.MetadataValue.float(avg_rate),
+            "values": dagster.MetadataValue.json(values),
+        }
+    )
+
+
+@dagster.asset(
+    partitions_def=HOURLY_PARTITION_DEFINITION,
+    ins={"exchange_rates": dagster.AssetIn(key=hourly_exchange_rates.key)},
+)
+def hourly_exchange_rates_in_clickhouse(
+    context: dagster.AssetExecutionContext,
+    exchange_rates: dict[str, Any],
+    cluster: dagster.ResourceParam[ClickhouseCluster],
+) -> dagster.MaterializeResult:
+    """
+    Stores exchange rates data in ClickHouse.
+    The base currency is always USD as per the table design.
+    """
+    # Extract data from the input
+    date_str = get_date_partition_from_hourly_partition(context.partition_key)
+
+    # Store the rates in ClickHouse
+    rows, values = store_exchange_rates_in_clickhouse(
+        context=dagster.build_op_context(), date_str=date_str, exchange_rates=exchange_rates, cluster=cluster
+    )
+
+    # Calculate some statistics for metadata
+    currencies_count = len(rows)
+    min_rate = min(row["rate"] for row in rows) if rows else 0.0
+    max_rate = max(row["rate"] for row in rows) if rows else 0.0
+    avg_rate = sum(row["rate"] for row in rows) / len(rows) if rows else 0.0
+
+    # Return the rows with metadata
+    return dagster.MaterializeResult(
+        metadata={
+            "date": dagster.MetadataValue.text(date_str),
+            "base_currency": dagster.MetadataValue.text("USD"),  # Always USD as per table design
+            "currencies_count": dagster.MetadataValue.int(currencies_count),
+            "min_rate": dagster.MetadataValue.float(min_rate),
+            "max_rate": dagster.MetadataValue.float(max_rate),
+            "avg_rate": dagster.MetadataValue.float(avg_rate),
+            "values": dagster.MetadataValue.json(values),
+        }
+    )
+
+
+# Create jobs from the assets
+daily_exchange_rates_job = dagster.define_asset_job(
+    name="daily_exchange_rates_job",
+    selection=[daily_exchange_rates.key, daily_exchange_rates_in_clickhouse.key],
+    tags={"owner": JobOwners.TEAM_WEB_ANALYTICS.value},
+)
+
+hourly_exchange_rates_job = dagster.define_asset_job(
+    name="hourly_exchange_rates_job",
+    selection=[hourly_exchange_rates.key, hourly_exchange_rates_in_clickhouse.key],
+    tags={"owner": JobOwners.TEAM_WEB_ANALYTICS.value},
+)
+
+
+# Create daily/hourly schedules with different cron schedules
+@dagster.schedule(
+    job=daily_exchange_rates_job,
+    cron_schedule="0 1 * * *",  # Run at 1:00 AM every day
```

Comment:
because a start of hour is so special, many customers sent us queries then, it's less alert prone to run it other time, can you make it some random,e.g. 1:37?

Reply:
Yeah, no prob at all, will change it to 0:28, just to make it run slightly earlier